homeassistant/components/stream/hls.py

```diff
@@ -49,11 +55,19 @@ async def handle(self, request, stream, sequence):
         """Return m3u8 playlist."""
         track = stream.add_provider("hls")
         stream.start()
-        # Wait for a segment to be ready
-        if not track.segments and not await track.recv():
+        if not track.sequences and not await track.recv(timeout=STREAM_TIMEOUT):
+            return web.HTTPNotFound()
+        # We actually need to see two segments before the first one has a duration
+        if len(track.sequences) == 1 and not await track.recv(timeout=STREAM_TIMEOUT):
             return web.HTTPNotFound()
-        headers = {"Content-Type": FORMAT_CONTENT_TYPE["hls"]}
-        return web.Response(body=self.render(track).encode("utf-8"), headers=headers)
+        response = web.Response(
+            body=self.render(track).encode("utf-8"),
+            headers={
+                "Content-Type": FORMAT_CONTENT_TYPE["hls"],
+            },
+        )
+        response.enable_compression(web.ContentCoding.gzip)
```

Comment:
From 6.2.1 of the RFC:

>    HTTP servers SHOULD transfer text files -- such as Playlists and
   WebVTT segments -- using the "gzip" Content-Encoding if the client
   indicates that it is prepared to accept it.